docs/testplan/srv6/SRv6-phoenixwing-traffic-testplan-using-trex.md

```diff
@@ -0,0 +1,416 @@
+# Ether()/IPv6(dst=segment, nh=43)/IPv6ExtHdrSegmentRouting(addresses=[vpnsid], nh=4, segleft=1)/IP(dst = dst_ip, tos=(dscp<<2))/UDP()
+check_topo_recv_pkt_srh_te(ptfadapter, port=0, dst_ip="", dscp = 0, vpnsid = "", segment = "", no_packet = False)
+```
+
+#### test case example
+A typical test case example, which demonstrates how to use the Trex functions and PTF functions to check the SRv6 traffic.
+
+```
+    test_ipv4_dip = "192.168.0.1"
+    result = trex_run(test_ipv4_dip, dscp = 4, duration = 10)
+    check_list = {"ptf_tot_rx": 10000, "ptf_tot_tx": 10000, "P1_tx_to_PE1": 2500, "P1_tx_to_PE2": 2500, "P2_tx_to_PE1": 2500, "P2_tx_to_PE2": 2500}
+    pytest_assert(thresh_check(result, check_list))
+
+    check_topo_recv_pkt_raw(ptfadapter, port=ptf_port_for_backplane, dst_ip=test_ipv4_dip, dscp = 4)
+    check_topo_recv_pkt_vpn(ptfadapter, port=ptf_port_for_P1_to_PE1, dst_ip=test_ipv4_dip, dscp = 4, vpnsid = NNI_ANY_PUBLIC_TC2)
+    check_topo_recv_pkt_srh_te(ptfadapter, port=ptf_port_for_PE3_to_P2, dst_ip=test_ipv4_dip, dscp = 4, vpnsid = NNI_ANY_PUBLIC_TC2, segment = SEGMENT_LIST_P2_P1)
+    reset_topo_pkt_counter(ptfadapter)
+```
+
+## Test Plan
+Previous testplan doc "SRv6-phoenixwing-ptf-testplan.md" has detailed the SRv6 test scenarios including SRv6 VPN and SRv6 Policy. In this testplan, We will add Trex traffic check in all these scenarios. The traffic check includes: packet loss, packet encapsulation, traffic ecmp result, and traffic convergence.
+
+### SRv6 VPN
+
+#### SRv6 VPN for single homing
+Publish VRF routes from PE1 to PE3, and run TRex traffic from PE3.
+
+- no packet loss: for example run trex for 10s and check {"ptf_tot_rx": 10000, "ptf_tot_tx": 10000} 
+- traffic ecmp from ingress PE to P: check {'PE3_tx_to_P2': 5000, 'PE3_tx_to_P4': 5000}
+- traffic ecmp from egress P to PE: check {'P1_tx_to_PE1': 5000, 'P1_tx_to_PE2': 0, 'P3_tx_to_PE1': 5000, 'P3_tx_to_PE2': 0}
+- packet encapsulation check: from PE3 to P2/P4, is IPinIPv6 packet with correct vpn sid; from P1/P3 to PE1, is IPinIPv6 packet with vpn correct sid
+- if send a single stream to PE3, the packet can only be found on a single path
+
+#### SRv6 VPN for Dual homing
+Publish VRF routes from PE1 and PE2 to PE3, and TRex run traffic from PE3.
+
+- no packet loss: for example run trex for 10s and check {"ptf_tot_rx": 10000, "ptf_tot_tx": 10000} 
+- traffic ecmp from ingress PE to P: check {'PE3_tx_to_P2': 5000, 'PE3_tx_to_P4': 5000}
+- traffic ecmp from egress P to PE: check {'P1_tx_to_PE1': 2500, 'P1_tx_to_PE2': 2500, 'P3_tx_to_PE1': 2500, 'P3_tx_to_PE2': 2500}
+- packet encapsulation check: from PE3 to P2/P4, is IPinIPv6 packet with correct vpn sid; from P1/P3 to PE1/PE2, is IPinIPv6 packet with vpn correct sid
+- if send a single stream to PE3, the packet can only be found on a single path
+
+#### Dual homing - IGP local failure case
+Start trex stream and wait 5 seconds; Shut down port between P2 and PE3; wait 5 seconds and Stop Trex stream.
```

Comment:
Please mention traffic end points for the test cases to visualize traffic load balancing in failure cases .

Reply:
Hi, All traffic end points are on PEs. Ingress PE3 is always to encap srv6 packets, and  Egress PE1/PE2 is to decap srv6 packets.

Reply:
added more images to show traffic load balancing

Reply:
@forrestchu - can this be presented on the next sonic-mgmt subgroup meeting?

Reply:
> @forrestchu - can this be presented on the next sonic-mgmt subgroup meeting?

that would be great, roy. But currently I don't get a meeting invitation, or should I schedule an appointment with the meeting owner?